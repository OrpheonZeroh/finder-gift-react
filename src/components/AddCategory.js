import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories } ) => {

    const [inputValue, setinputValue] = useState('') //si lo dejas en blanco, queda como indefinido

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e)=> {

        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories(cats=> [inputValue, ...cats]);
            setinputValue('')
        }

        // console.log('Submit hecho')
        

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={handleInputChange}
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}