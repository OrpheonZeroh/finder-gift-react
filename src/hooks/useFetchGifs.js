import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((imgs) => {
      console.log(imgs);

      setState({
        data: imgs,
        loading: false,
      });
    });
  }, [category]);

  // useEffect( ()=> {
  //     getGifs(category)
  //         // .then( imgs => setImages(imgs))
  //         .then( setImages );
  // }, [ category ])

  // setTimeout(() => {
  //     setState({
  //         data: [1,2,3,4,5,6],
  //         loading: false
  //     })
  // }, 3000)

  return state; // {data:[], loading:true}
};
