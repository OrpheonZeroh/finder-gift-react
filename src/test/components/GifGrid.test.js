import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('pruebas con < GifGrid />', () => {

    const category = 'One Punch';


    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    // const category = 'Drango Ball'
    
    
    // test('debe renderizr correctamente el componente', () => {
  
    //     useFetchGifs.mockReturnValue({
    //         data: [],
    //         loading: true,
    //     });

    //     const wrapper = shallow(<GifGrid category={ category }/>)
    //     expect( wrapper ).toMatchSnapshot()

    // });

    // test('debe de mostrar item cuando se cargan imagenes usefechtGifs', () => {
        
    //     const gifs = [{
    //         id: '1',
    //         title: 'Epalee',
    //         url: 'https://localhost/epalee'
    //     }]

    //     useFetchGifs.mockReturnValue({
    //         data: gifs,
    //         loading: false,
    //     });

    //     expect( wrapper ).toMatchSnapshot()

    // });

});