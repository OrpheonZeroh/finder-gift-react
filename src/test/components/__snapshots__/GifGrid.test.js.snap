// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con < GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
